src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzBqJobUserStep.java

```diff
@@ -44,14 +44,9 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     List<String> policyEmails =
         new ArrayList<>(List.of(policyMap.get(IamRole.STEWARD), policyMap.get(IamRole.READER)));
 
-    if (sourceDataset.isInheritSteward()) {
-      Map<IamRole, String> sourceDatasetPolicyMap =
-          workingMap.get(
-              SnapshotWorkingMapKeys.SOURCE_DATASET_POLICY_MAP, new TypeReference<>() {});
-      // Allow the custodian to make queries in this project.
-      policyEmails.add(sourceDatasetPolicyMap.get(IamRole.CUSTODIAN));
-    }
-    // The underlying service provides retries so we do not need to retry this operation
+    addSourceDatasetPolicyEmailsIfInheritSteward(workingMap, policyEmails, sourceDataset);
```

Comment:
If `addSourceDatasetPolicyEmailsIfInheritSteward` returned a `List` you could generate the List without needing to make a mutable list.
```
    List<String> policyEmails =
        Stream.concat(
                Stream.of(policyMap.get(IamRole.STEWARD), policyMap.get(IamRole.READER)),
                addSourceDatasetPolicyEmailsIfInheritSteward(workingMap, sourceDataset).stream())
            .toList();
```

Reply:
I think keeping them as separate calls helps with readability, and it is then more consistent across the three steps that call this method.